modules/api/pkg/resources/machine/common.go

```diff
@@ -619,14 +619,23 @@ func GetKubevirtProviderConfig(cluster *kubermaticv1.Cluster, nodeSpec apiv1.Nod
 		},
 	}
 
-	if dc.Spec.Kubevirt.EnableDedicatedCPUs {
+	// defaulting for both flags regarding virtual machine cpu assignment:
+	// - EnableDedicatedCPUs defaults (false) to .resources.requests/limits
+	// - UsePodResourcesCPU defaults (false) to .domain.cpu
+	// as long as we have the deprecated and new flag we need to take care about 4 cases
+	// 1) old flag is set to true and new flag is set to false => .domain.cpu will be used in kubevirt vm
+	// 2) old flag is set to true and new flag is set to true => .domain.cpu will be used in kubevirt vm
+	// 3) old flag is set to false and new flag is set to false => .domain.cpu will be used in kubevirt vm
+	// 4) old flag is set to false and new flag is set to true => .resources.requests/limits will be used
+	// in kubevirt vm
```

Comment:
```suggestion
	// defaulting for both flags regarding virtual machine cpu assignment:
	// - EnableDedicatedCPUs defaults (false) to .resources.requests/limits
	// - UsePodResourcesCPU defaults (false) to .domain.cpu
	// as long as we have the deprecated and new flag we need to account for 4 cases:
	// 
	// | Old Flag (EnableDedicatedCPUs) | New Flag (UsePodResourcesCPU) | Source for KubeVirt VM CPU |
	// |---------|---------|---------|
	// | true | false | .domain.cpu |
	// | true | true | .domain.cpu |
	// | false | false | .domain.cpu |
	// | false | true | .resources.requests/limits |
```

If this improved comment can help in readability otherwise please ignore this comment.
thx